app/components/banners/maintenance_component.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Banners
+  class MaintenanceComponent < ViewComponent::Base
+    attr_reader :wide_container_view
+
+    # Set the following constants to the appropriate dates and times to enable the upcoming maintenance warning.
+    #
+    # date - required (for rendering)
+    # start_time - optional
+    # end_time - optional
+    DATE       = nil # Date.new(2021, 8, 4)
+    START_TIME = nil # Time.new(2021, 8, 4, 9, 0, 0)
+    END_TIME   = nil # Time.new(2023, 8, 4, 17, 0, 0)
+
+    def initialize(wide_container_view: false)
+      @wide_container_view = wide_container_view
+    end
+
+    def render?
+      date.present?
```

Comment:
I wonder if we should allow this to be toggled via a feature flag instead? So we don't have to deploy to reset date to `nil`?

Reply:
Agree, I know we need a date but switching on/off via a `FeatureFlag` would be great

Reply:
That's a fair point, I only just saw this and have just pushed another change which is that the banner goes away automatically once the date it is set to passes.